pallets/order-book/src/lib.rs

```diff
@@ -233,6 +227,40 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	/// Storage of valid order pairs.
+	/// Stores:
+	///  - key1 -> AssetIn
+	///  - key2 -> AssetOut
+	///
+	/// Stores the minimum `buy_amount` of `asset_in` when buying
+	/// with `asset_out`
+	#[pallet::storage]
+	pub type TradingPairInOut<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		T::AssetCurrencyId,
+		Twox64Concat,
+		T::AssetCurrencyId,
+		T::Balance,
+		ResultQuery<Error<T>::InvalidTradingPair>,
+	>;
+
+	/// Storage of valid order pairs.
+	/// Stores:
+	///  - key1 -> AssetOut
+	///  - key2 -> AssetIn
+	// NOTE: Mirror storage for easier querability.
+	#[pallet::storage]
+	pub type TradingPairOutIn<T: Config> = StorageDoubleMap<
```

Comment:
I am failing to see when we would need this mirrored storage. IMO it can be replaced by calling `TraidngPairInOut::<T>::(out, in)`?

Reply:
Frontend. But most not be used. Can also remove

Reply:
If we plan on treating trading pairs as directed edges of a graph where each node is a currency id, then we actually should keep this storage. When I commented, I wrongly assumed the graph to be undirected.

Reply:
You are right. Removing it.